server/src/middleware/validate_token.rs

```diff
@@ -19,76 +20,227 @@ pub async fn validate_token(
         .unwrap()
         .clone()
         .into_inner();
+
+    let validation_status = validate(&token, maybe_validator, &token_cache, req.path()).await;
+
+    if let Ok(_) = validation_status {
+        Ok(srv.call(req).await?.map_into_left_body())
+    } else {
+        Ok(req
+            .into_response(HttpResponse::Forbidden().finish())
+            .map_into_right_body())
+    }
+}
+
+async fn validate(
+    token: &EdgeToken,
+    maybe_validator: Option<&Data<impl TokenRegister>>,
+    token_cache: &DashMap<String, EdgeToken>,
+    path: &str,
+) -> Result<(), EdgeError> {
+    if token.status == TokenValidationStatus::Trusted {
```

Comment:
This is the magic here - pure short circuit if the token is trusted. This function is effectively the same as it was, just decomposed so we can test it